### ISSUE TITLE
Refactor Curaçao as NL/CW subdivision

### ISSUE THREAD

**Issue by arkid15r** [vacanza/python-holidays#1852]
### Description

Make sure Curaçao is available as a subdiv of Netherlands (based on https://www.iso.org/obp/ui/en/#iso:code:3166:CW).

### Related Issues

N/A

### Proposed Solution

Move the `CW` holidays logic to `NL` class. Refactor Curaçao logic to be based on `NL`'s subdiv `CW`

### Additional context

N/A

**Comment by KJhellico** [comment]
It seems logical to do the same for [Aruba](https://www.iso.org/obp/ui/en/#iso:code:3166:AW).

**Comment by arkid15r** [comment]
> It seems logical to do the same for [Aruba](https://www.iso.org/obp/ui/en/#iso:code:3166:AW).

Absolutely! I've updated the issue. Thanks!

**Comment by PPsyrius** [comment]
Quick question here: Isle of Man, Jersey, and Guernsey were previously callable as Great Britain subdivisions before 2007: https://en.wikipedia.org/wiki/ISO_3166-2:GB as `GB-IOM,` `GB-GSY,` and `GB-JSY` respectively. Should these entities get the same treatment, too?

Ideally, we should have ways to call up the appropriate holiday codes rather than folding them into the main entities. That would be great here.

**Comment by PanderMusubi** [comment]
@arkid15r if they have extra or other holidays on Aruba and Curaçao, please also add it to https://commenthol.github.io/date-holidays/#nl/2025/NL Thanks.

**Comment by arkid15r** [comment]
> [@arkid15r](https://github.com/arkid15r) if they have extra or other holidays on Aruba and Curaçao, please also add it to https://commenthol.github.io/date-holidays/#nl/2025/NL Thanks.

Yes, we normally do that (`CW` already has multiple categories supported) -- feel free to provide any official sources you're aware of.
Thank you!
